programming_examples/vision/color_threshold/color_threshold.py

```diff
@@ -61,16 +61,21 @@ def color_threshold(dev, width, height):
             )
         )
 
-    # Task for the core to perform
-    def core_fn(of_in, of_out, my_rtp, threshold_fn):
-        elemIn = of_in.acquire(1)
-        elemOut = of_out.acquire(1)
+    # Create a barrier to synchronize individual workers with the runtime sequence
+    barrier = WorkerRuntimeBarrier()
 
-        # RTPs written from the instruction stream must be read right before the kernel
-        # after the ObjectFIFO acquires
+    # Task for the core to perform
+    def core_fn(of_in, of_out, my_rtp, threshold_fn, barrier):
+        # RTPs written from the instruction stream must be synchronized with the runtime sequence
+        # This may be done explicitly through the usage of a barrier or implicitly though the usage of ObjectFIFO acquire
```

Comment:
I agree that previously examples would piggy back off of the objectfifo's synchronization, however, I wouldn't suggest that as a correct way of waiting for RTPs. Using your new ops is the correct way to do it. The programming guide should be updated with this expected practice as well.